spec/controllers/marks_controller_spec.rb

```diff
@@ -0,0 +1,157 @@
+require 'rails_helper'
+
+RSpec.describe MarksController, type: :controller do
+    let(:new_spy) { build(:spy) }
+    let(:new_mark) { build(:mark) }
+    let(:prez) { build(:mark, :prez) }
+
+  describe "GET #results" do
+    context "logged in" do
+      context "twitter" do
+        it "renders the results template" do
+          new_spy.save
+          session[:spy_id] = new_spy.id
+          get :results, provider: "twitter", username: "loganmeetsworld"
+          expect(response).to render_template :results
+        end
+      end
+
+      context "vimeo" do
+        it "renders the results page" do
+          new_spy.save
+          session[:spy_id] = new_spy.id
+          get :results, provider: "vimeo", username: "loganmeetsworld"
+          expect(response).to render_template :results
+        end
+      end
+    end
+
+    context "not logged in" do
+      context "twitter" do
+        it "redirects to the home page" do
+          get :results, provider: "twitter"
+          expect(response).to redirect_to root_path
+        end
+
+        it "adds a flash error" do
+          get :results, provider: "twitter"
+          expect(flash[:error]).to_not be nil
+        end
+      end
+
+      context "vimeo" do
+        it "redirects to the home page" do
+          get :results, provider: "vimeo"
+          expect(response).to redirect_to root_path
+        end
+
+        it "adds a flash error" do
+          get :results, provider: "vimeo"
+          expect(flash[:error]).to_not be nil
+        end
+      end
+    end
+  end
+
+  describe "GET #index" do
+    it "renders the index page" do
+      new_spy.save
+      session[:spy_id] = new_spy.id
+      get :index
+      expect(response).to render_template :index
+    end
+  end
+
+  describe "POST #vimeo_subscribe" do
+    it "redirects to the marks index page" do
+      new_spy.save
+      session[:spy_id] = new_spy.id
+      post :vimeo_subscribe, name: "hi"
+      expect(response).to redirect_to marks_path
+    end
+
+    it "creates a new mark" do
+      new_spy.save
+      session[:spy_id] = new_spy.id
+      expect{ post :vimeo_subscribe, name: "hi" }.to change(Mark, :count).by(1)
+    end
+
+    it "doesn't create a new mark if one already exists" do
+      new_spy.save
+      session[:spy_id] = new_spy.id
+      post :vimeo_subscribe, name: "hi"
+
+      expect{ post :vimeo_subscribe, name: "hi" }.to change(Mark, :count).by(0)
+
+    end
+
+    it "adds mark to spies_marks when one exists in db vimeo" do 
+      new_spy.save
+      new_mark.save
+      new_spy.marks << new_mark
+      session[:spy_id] = new_spy.id
+      delete :destroy, id: new_mark.id
+      post :vimeo_subscribe, name: "jeffdesom"
+
+      expect(new_spy.marks.size).to eq 1
+    end
+  end
+
+  describe "POST #twitter_subscribe" do
+    it "redirects to the marks index page" do
+      new_spy.save
+      session[:spy_id] = new_spy.id
```

Comment:
You seem to use
```new_spy.save
session[:spy_id] = new_spy.id```
a lot in this and many other describe blocks.
This would be something you could potentially extract out into a `beforeEach`.